### PR TITLE
Update GetPrint doc

### PR DESCRIPTION
**Pull request by pblottiere**

### CODE REVIEW

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 1]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
     ...
     </WMS_Capabilities>
 
-The client has now the information to request a print output::
 
-    http://myserver.com/cgi/qgis_mapserv.fcgi?...&REQUEST=GetPrint&TEMPLATE=Druckzusammenstellung 1&map0:EXTENT=xmin,ymin,xmax,ymax&map0:ROTATION=45&FORMAT=pdf&DPI=300
+SERVICE
+^^^^^^^
+
+This parameter has to be ``WMS`` in case of the **GetMap** request.
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
This parameter has to be ``WMS``.
```

Comment (havatv) [reply source/docs/user_manual/working_with_ogc/server/services.rst]: Or have I misunderstood?

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 2]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
...
-* **<map_id>:SCALE** Sets a map scale to a layout map item. This is useful to ensure
-  scale based visibility of layers and labels even if client and server may
-  have different algorithms to calculate the scale denominator
-* **<map_id>:LAYERS**, **<map_id>:STYLES** possibility to give layer and styles
-  list for layout map item (useful in case of overview maps which should have only
-  a subset of layers)
+REQUEST
+^^^^^^^
+
+This parameter is ``GetPrint`` in case of the **GetPrint** request.
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
This parameter has to be ``GetPrint`` for the **GetPrint** request.
```

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 3]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
...
+
+
+TEMPLATE
+^^^^^^^^
+
+This parameter allows to specify the name of the layout template to use for
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
This parameter can be used to specify the name of a layout template to use for
```

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 4]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
...
+printing.
+
+
+FORMAT
+^^^^^^
+
+This parameter may be used to specify the format of map image. Available
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
This parameter specifies the format of map image. Available
```

Comment (havatv) [reply source/docs/user_manual/working_with_ogc/server/services.rst]: The parameter is mandatory

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 5]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
...
+- ``svg``
+
+If the ``FORMAT`` parameter is different from one of these values, then an
+exception is returned.
+
+
+ATLAS_PK
+^^^^^^^^
+
+This parameter allows to activate the Atlas rendering by indicating which
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
This parameter allows activatation of Atlas rendering by indicating which
```

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 6]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
...
+features we want to print. In order to retrieve an atlas with all features,
+the ``*`` symbol may be used (according to the maximum number of feature
+allowed in the project configuration).
+
+When ``FORMAT`` is ``pdf``, then we have a PDF document with one feature per
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
When ``FORMAT`` is ``pdf``, a PDF document with one feature per
```

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 7]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
...
+page. For all other formats, only the first page is returned.
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
page will be returned. For all other formats, only the first page is returned.
```

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 8]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
...
+
+
+mapX:EXTENT
+^^^^^^^^^^^
+
+This parameter allows to give the extent for a layout map item as
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
This parameter specifies the extent for a layout map item as
```

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 9]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
...
+xmin,ymin,xmax,ymax.
+
+
+mapX:ROTATION
+^^^^^^^^^^^^^
+
+This parameter allows too indicate the map rotation in degrees.
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
This parameter specifies the map rotation in degrees.
```

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 10]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
...
+
+
+mapX:GRID_INTERVAL_X
+^^^^^^^^^^^^^^^^^^^^
+
+This parameter allows to indicate the grid line density in X direction.
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
This parameter specifies the grid line density in X direction.
```

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 11]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
...
+
+
+mapX:GRID_INTERVAL_Y
+^^^^^^^^^^^^^^^^^^^^
+
+This parameter allows to indicate the grid line density in Y direction.
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
This parameter specifies the grid line density in the Y direction.
```

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 12]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
...
+
+
+mapX:SCALE
+^^^^^^^^^^
+
+This parameter allows to set a map scale to a layout map item. This is useful
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
This parameter specifies the map scale for a layout map item. This is useful
```

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 13]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
...
+to ensure scale based visibility of layers and labels even if client and server
+may have different algorithms to calculate the scale denominator.
+
+
+mapX:LAYERS
+^^^^^^^^^^^
+
+This parameter allows to specify layers for a specific layout map item. See
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
This parameter specifies the layers for a layout map item. See
```

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 14]
```diff
@@ -585,22 +651,101 @@ they are listed as possible print templates:
...
+:ref:`See GetMap <qgisserver-wms-getmap>`: for more information on this
+parameter.
+
+
+mapX:STYLES
+^^^^^^^^^^^
+
+This parameter allows to specify a layers' styles defined in a specific
```
Comment (havatv) [inline source/docs/user_manual/working_with_ogc/server/services.rst]:
```suggestion
This parameter specifies the layers' styles defined in a specific
```

**Comment by pblottiere** [comment]
Hi @havatv,

I think a took into account all your comments.

Thanks for your review!

File: source/docs/user_manual/working_with_ogc/server/services.rst [hunk 15]
```diff
@@ -585,22 +651,100 @@ they are listed as possible print templates:
...
+FORMAT
+^^^^^^
+
+This parameter specifies the format of map image. Available values are:
+
+- ``jpg``
+- ``jpeg``
+- ``image/jpeg``
+- ``png``
+- ``svg``
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_ogc/server/services.rst]: Should pdf be listed here?

**Comment by pblottiere** [comment]
> It also formats the section according to the new structuring, which is a great move. The bad side is that atlas getprint is a 3.6 feature (it will not be backported, right?) while we are still on 3.4 milestone for docs. Could we do that in two steps:

> 1. merging only the refactoring
> 2. and keeping aside a PR for atlas_pk we will merge when time comes?

> Or do I look too much strict?

No, you're perfectly right :+1:. I'm going to do that.

File: source/docs/user_manual/working_with_ogc/server/services.rst (hunk 5 above)
Comment (havatv) [reply source/docs/user_manual/working_with_ogc/server/services.rst]: Yes!

File: source/docs/user_manual/working_with_ogc/server/services.rst (hunk 6 above)
Comment (havatv) [reply source/docs/user_manual/working_with_ogc/server/services.rst]: Is that not included in the next suggestion?
("page will be returned")

**Comment by pblottiere** [comment]
Can you wait a little before merging this PR? I think that I need to update the ``FORMAT`` parameter.

File: source/docs/user_manual/working_with_ogc/server/services.rst (hunk 6 above)
Comment (DelazJ) [reply source/docs/user_manual/working_with_ogc/server/services.rst]: Actually it confuses me more.
#### 1. For all other formats, only the *first page* is returned.
Is that possible to have a two pages layout on server? If yes, with jpg, I'd lose the second page? Am I right?

#### 2. "When ``FORMAT`` is ``pdf``, a PDF document with *one feature per page* is output. 
For me (i'm not server user but it'd be inline with the desktop) it's more a matter of file(s) per feature than page per feature. What I naively thought would be the behavior is:
"When ``FORMAT`` is ``pdf``,  a single PDF document combining each feature page(s) is output. For all other formats, each feature is returned in separate file(s)." 

The limitation with pdf on server would be that I can not output separate pdf files for each feature.
note: the concerned part is no longer in this PR.

**Comment by pblottiere** [comment]
Feel free to merge, thanks!

**Pull request merged**